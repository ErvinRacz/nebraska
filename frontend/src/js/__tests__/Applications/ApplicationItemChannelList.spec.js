import {render} from '@testing-library/react';
import React from 'react';
import ApplicationItemChannelsList from '../../components/Applications/ApplicationItemChannelsList.react';

describe('Application Channel List ', () => {
  it('should render no application channel for empty channels', () => {
    const {container} = render(<ApplicationItemChannelsList channels={[]}/>);
    expect(container.firstChild).toBeEmpty();
  });
  it('should render channel item if there are channels', () => {
    const channels = [{
      'id': 'ABC',
      'name': 'main',
      'color': '#777777',
      'created_ts': '2018-10-16T21:07:56.819939+05:30',
      'application_id': 'DEF',
      'package_id': 'XYZ',
      'package': {'id': 'PACK_ID', 'type': 4,
                  'version': '1.11.3', 'url': 'https://github.com/kinvolk',
                  'filename': '', 'description': '', 'size': '', 'hash': '', 'created_ts': '2019-07-18T20:10:39.163326+05:30',
                  'channels_blacklist': null, 'application_id': 'df1c8bbb-f525-49ee-8c94-3ca548b42059', 'flatcar_action': null, 'arch': 0},
      'arch': 0}];
    const {asFragment} = render(<ApplicationItemChannelsList channels={channels}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
