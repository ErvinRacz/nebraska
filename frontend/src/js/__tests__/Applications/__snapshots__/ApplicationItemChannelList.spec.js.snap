// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Channel List  should render channel item if there are channels 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
    >
      <div
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-colorAvatar-220 makeStyles-colorAvatar-221 MuiAvatar-colorDefault"
          >
            m
          </div>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            main
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            1.11.3 (ALL)
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
